var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -676,9 +674,9 @@ def suitesparse_components(self):
     @property
     def sundials_components(self):
         """Return the SUNDIALS components needed by MFEM."""
-        sun_comps = 'arkode,cvode,nvecserial,kinsol'
+        sun_comps = 'arkode,cvodes,nvecserial,kinsol'
         if '+mpi' in self.spec:
-            sun_comps += ',nvecparhyp,nvecparallel'
+            sun_comps += ',nvecparhyp,nvecparallel,nvecmpiplusx'
```

Comment:
The recent change in mfem's `master` branch is that `nvecparhyp` is no longer needed and instead `nvecmpiplusx` is required. So adding both here should work with all versions (while linking with one redundant library) as long as `nvecparhyp` and `nvecmpiplusx` were always both available in sundials -- I think `nvecmpiplusx` may be newer and not in all versions where `nvecparhyp` is available -- @gardner48 is this correct?

Having some logic to switch between `nvecparhyp` and `nvecmpiplusx` based on the mfem version may be nice but probably not strictly necessary.

Reply:
That's right, `nvecmpiplusx` was added in sundials 5.0.0 and is always built when sundials is configured with MPI enabled.

`nvecparhyp` was used in an older version of the mfem interface to sundials but some time before mfem 4.1 it was no longer used in the interfacing (I'm not sure when that change happened though). This vector is only built when sundials is configured with MPI and hypre enabled.

Reply:
Okay, in order to continue to support older versions of mfem with older versions of sundials, I think we should add an if statement here: for mfem versions `@4.2:`, i.e. 4.2 (upcoming) and up (which will include `master`) we want to do `sun_comps += ',nvecparallel,nvecmpiplusx'`, otherwise `sun_comps += ',nvecparhyp,nvecparallel'`. Something like this:
```python
             if self.spec.satisfies('@4.2:'):
                 sun_comps += ',nvecparallel,nvecmpiplusx'
             else:
                 sun_comps += ',nvecparhyp,nvecparallel'
```
I think the fact that v4.2 is not yet defined as a version is okay.